api_bonus.yaml

```diff
@@ -0,0 +1,144 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: Bonus API
+host: localhost
+basePath: /api/v1
+schemes:
+  - https
+security:
+  - Bearer: []
+paths:
+  "/bonus/vacanze/request":
+    post:
+      operationId: startBonusEligibilityCheck
+      summary: Start bonus eligibility check (ISEE)
+      responses:
+        '202':
+          description: Request accepted.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          headers:
+            Location:
+              type: string
+              description: |-
+                Location (URL) of created request resource.
+                A GET request to this URL returns the request status and details.
+          examples:
+            application/json:
+              id: 01BX9NSMKVXXS5PSP2FATZMYYY
+
+    get:
+      operationId: getBonusEligibilityCheck
+      summary: Get eligibility (ISEE) check information for user's bonus
+      responses:
+        '202':
+          description: Processing request.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          examples:
+            application/json:
+              id: 01BX9NSMKVXXS5PSP2FATZMYYY
+        '200':
+          description: Request processed.
+          schema:
+            type: object
+            properties:
+              id:
+                type: string
+                description: The identifier of the created request.
+          examples:
+            application/json:
+              # REQUEST (isee): NO_DATA, ELIGIBLE, INELIGIBLE, NOT_FOUND
+              status: ELIGIBLE
+              members:
+                - name: "Mario"
+                  surname: "Rossi"
+                - name: "Piero"
+                  surname: "Rossi"
+                - name: "Giulia"
+                  surname: "Rossi"
+              max_amount: 50000
+              tax_benefit: 3000
+
+    delete:
```

Comment:
this also should have the task ID in the path

Reply:
not if the task id equals the user's fiscal code

Reply:
@gunzip 
Since **eligibility** payload has always an **id** but could not have the other fields (status, family_members, etc etc) because its check is pending, can we consider to make **id** as required field and add a new field (ex **check_result**) containing the result data (status, family_members, etc etc) as optional object?

Reply:
every field is required (you'll get two different schemas, one for 202 and another one for 200)